### PR TITLE
e2e: hide verbose usage info for error message

### PR DESCRIPTION
**Pull request by ZhouXing19**
Release note: None

### CODE REVIEW

**Comment by ZhouXing19** [comment]
@ryanluu12345 This PR might help with the unhide --non-interactive one

**Review by ryanluu12345 — APPROVED** [review]

File: e2e/datadriven_test.go [hunk 1]
```diff
@@ -139,3 +139,8 @@ func redactLogs(s string) string {
 	const fileServerStartingPattern = `(?m)^[^\n]*starting file server[^\n]*\n?`
 	return regexp.MustCompile(fileServerStartingPattern).ReplaceAllString(res, "")
 }
+
+func extractErrorOnlyFromMoltRes(s string) string {
```
Comment (ryanluu12345) [inline e2e/datadriven_test.go:143]: I think this is a change that will set us up for easier maintenance of the e2e tests, especially when we change flags. Thanks!

File: e2e/datadriven_test.go [hunk 2]
```diff
@@ -139,3 +139,8 @@ func redactLogs(s string) string {
...
+	const errPattern = `Error: [^\n]+\n`
```
Comment (ryanluu12345) [inline e2e/datadriven_test.go:144]: What happens in the case where there are multi-line errors? A case I can think of is if the underlying error has multiple lines (`\n` is a valid substring inside). I think it's worthwhile to add a test for the case where that happens.

**Comment by ryanluu12345** [comment]
> @ryanluu12345 This PR might help with the unhide --non-interactive one

Ahh just read this part. You may need to rebase since I merged that `--non-interactive` change already

**Pull request merged**